File: data/readable_format/fortran/file1841.f90
PROGRAM P
  IMPLICIT NONE
  INTEGER :: status
  
  PRINT *, "Calling S subroutine"
  CALL S(status)
  
  IF (status == 0) THEN
     PRINT *, "S executed with STOP 0 behavior."
  ELSE
     PRINT *, "S did not execute as expected."
  END IF

  STOP 1
CONTAINS

   SUBROUTINE S(status)
      INTEGER, INTENT(OUT) :: status
      PRINT *, "Inside S subroutine, simulating STOP 0."
      status = 0  ! Simulate STOP 0 behavior by setting status
      ! Normally, we'd use STOP 0 here, but for demonstration, we're simulating it
   END SUBROUTINE S
END PROGRAM P

File: data/readable_format/cpp/file1841.cpp
#include <cstdlib> // For exit() and EXIT_SUCCESS, EXIT_FAILURE
#include <iostream> // For std::cout

// Declaration of function S() before use
int S();

int main() {
    if (S() == 0) {
        std::cout << "S() executed with expected behavior." << std::endl;
    } else {
        std::cout << "S() did not execute as expected." << std::endl;
    }

    // Simulate the STOP 1 behavior with a return value
    return 1; // This would be the place to simulate STOP 1
}

// Definition of subroutine S
int S() {
    // Instead of exiting, return a value to simulate STOP 0
    return 0;
}